infra.go

```diff
@@ -56,7 +56,36 @@ type InfrastructureResourceInput struct {
 	Data         JSON                                 `json:"data,omitempty" yaml:"data" scalar:"true"`
 }
 
-func (client *Client) CreateInfrastructure(input InfrastructureResourceInput) (*InfrastructureResource, error) {
+type InfraProviderInput struct {
```

Comment:
These new input types don't exist in our API but they provide a WAY better customer UX when using the library.  See the code in the CLI and Terraform as implementation examples.